distributed/utils.py

```diff
@@ -1052,19 +1054,48 @@ def ensure_memoryview(obj):
         return mv
 
 
-def open_port(host=""):
-    """Return a probably-open port
+def open_port(host="", port=0):
```

Comment:
the version of ephemeral-port-reserve seems to be pretty close to what we had but much more sophisticated and likely better tested. Since `open_port` is something that might've been used elsewhere I decided to vendor the code.

Reply:
That version didn't work on windows so I reverted to the old one.